goal_src/levels/common/baseplat.gc

```diff
@@ -5,33 +5,564 @@
+    (set! (-> v1-2 frame-num) 0.0)
+    )
+   (suspend)
+   (dummy-47 (-> self root-override))
+   (ja-post)
+   (while #t
+    (when
+     (and
+      *target*
+      (>=
+       (-> self open-distance)
+       (vector-vector-distance
+        (-> self root-override trans)
+        (-> *target* control trans)
+        )
+       )
+      )
+     (dummy-26 self)
+     (if
+      (and
+       (not (-> self locked))
+       (or
+        (and
+         (-> self entity)
+         (logtest?
+          (-> self entity extra perm status)
+          (entity-perm-status complete)
+          )
+         )
+        (let ((a1-1 (new 'stack-no-clear 'event-message-block)))
+         (set! (-> a1-1 from) self)
+         (set! (-> a1-1 num-params) 2)
+         (set! (-> a1-1 message) 'query)
+         (set! (-> a1-1 param 0) (the-as uint 'powerup))
+         (set! (-> a1-1 param 1) (the-as uint 3))
+         (send-event-function *target* a1-1)
+         )
+        (and
+         (-> self one-way)
+         (< (vector4-dot (-> self out-dir) (target-pos 0)) -8192.0)
+         )
+        )
+       )
+      (go-virtual door-opening)
+      )
+     (level-hint-spawn
+      (game-text-id zero)
+      (the-as string #f)
+      (the-as symbol (-> self entity))
+      *entity-pool*
+      0
+      )
+     )
+    (suspend)
+    )
+   (none)
+   )
+  )
+
+;; failed to figure out what this is:
+(defstate door-opening (eco-door)
+  :virtual #t
+  :event
+  eco-door-event-handler
+  :code
+  (behavior ()
+   (let
+    ((gp-0
+      (and
+       (not
+        (and
+         (-> self entity)
+         (logtest?
+          (-> self entity extra perm status)
+          (entity-perm-status complete)
+          )
+         )
+        )
+       (let ((a1-0 (new 'stack-no-clear 'event-message-block)))
+        (set! (-> a1-0 from) self)
+        (set! (-> a1-0 num-params) 2)
+        (set! (-> a1-0 message) 'query)
+        (set! (-> a1-0 param 0) (the-as uint 'powerup))
+        (set! (-> a1-0 param 1) (the-as uint 3))
+        (send-event-function *target* a1-0)
+        )
+       )
+      )
+     )
+    (if gp-0
+     (sound-play-by-name
+      (static-sound-name "blue-eco-on")
+      (new-sound-id)
+      1024
+      0
+      0
+      (the-as uint 1)
+      (-> self root-override trans)
+      )
+     )
+    (sound-play-by-name
+     (-> self open-sound)
+     (new-sound-id)
+     1024
+     0
+     0
+     (the-as uint 1)
+     (the-as vector #t)
+     )
+    (dummy-48 (-> self root-override))
+    (until (ja-done? 0)
+     (let ((a0-9 (-> self skel root-channel 0)))
+      (set!
+       (-> a0-9 param 0)
+       (the float (+ (-> a0-9 frame-group data 0 length) -1))
+       )
+      (set! (-> a0-9 param 1) (-> self speed))
+      (joint-control-channel-group-eval!
+       a0-9
+       (the-as art-joint-anim #f)
+       num-func-seek!
+       )
+      )
+     (if (and gp-0 (rand-vu-percent? 0.5))
+      (spawn-projectile-blue *target*)
```

Comment:
looks like the mysterious blue projectile is the blue eco effect that goes from Jak to the door that's being opened.